src/lib.rs

```diff
@@ -25,23 +26,27 @@ fn check_mut_ptr<T>(ptr: *mut T) -> Result<*mut T> {
     }
 }
 
-fn check_retcode(n: &Nfs, code: i32) -> Result<()> {
+fn check_retcode(ctx: *mut nfs_context, code: i32) -> Result<()> {
     if code < 0 {
-        Err(Error::new(ErrorKind::Other, n.get_nfs_error()?))
+        unsafe {
+            let err_str = nfs_get_error(ctx);
+            let e = CStr::from_ptr(err_str).to_string_lossy().into_owned();
+            Err(Error::new(ErrorKind::Other, e))
+        }
     } else {
         Ok(())
     }
 }
 
 pub struct Nfs {
-    context: *mut nfs_context,
+    context: Rc<*mut nfs_context>,
 }
 
 impl Drop for Nfs {
     fn drop(&mut self) {
         if !self.context.is_null() {
             unsafe {
-                nfs_destroy_context(self.context);
+                nfs_destroy_context(*self.context);
```

Comment:
This is the problem here - if you drop `Nfs` here it will unconditionally destroy the `nfs_context` even if there are still instances of `Rc<*mut nfs_context>`, so they'll be left with dangling pointers to freed memory.